Build error: cmake

cmake errors even with cxxflags set (spack compiled gcc@9.2.0)

`spack install cmake cxxflags="-std=c++11"`

---

### Steps to reproduce the issue

```console
spack install cmake cxxflags="-std=c++11"

2 errors found in build log:
     454    -- Checking if compiler supports needed C++17 constructs
     455    -- Checking if compiler supports needed C++17 constructs - yes
     456    -- Checking if compiler supports C++ make_unique
     457    -- Checking if compiler supports C++ make_unique - no
     458    -- Checking if compiler supports C++ unique_ptr

  >> 474    CMake Error at CMakeLists.txt:92 (message):
     475      The C++ compiler does not support C++11 (e.g.  std::unique_ptr).
```

### Platform and user environment

```commandline
$ uname -a 
Linux edi 3.10.0-1062.1.1.el7.x86_64 #1 SMP Fri Sep 13 22:55:44 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux

centOS 7.7
``` 

### Additional information

Possibly related https://github.com/spack/spack/issues/11223

I've hit the same error just now on a fresh pull from 'develop'.

```
2 errors found in build log:
     904    -- Checking if compiler supports needed C++14 constructs
     905    -- Checking if compiler supports needed C++14 constructs - no
     906    -- Checking if compiler supports C++ make_unique
     907    -- Checking if compiler supports C++ make_unique - no
     908    -- Checking if compiler supports C++ unique_ptr
     909    -- Checking if compiler supports C++ unique_ptr - no
  >> 910    CMake Error at CMakeLists.txt:92 (message):
     911      The C++ compiler does not support C++11 (e.g.  std::unique_ptr).
     912
     913
     914    -- Configuring incomplete, errors occurred!
     915    See also "/tmp/root/spack-stage/cmake-3.15.3-zelieiidcfymrkb2l4zd55
            pnmyrrsbnp/spack-src/CMakeFiles/CMakeOutput.log".
     916    See also "/tmp/root/spack-stage/cmake-3.15.3-zelieiidcfymrkb2l4zd55
            pnmyrrsbnp/spack-src/CMakeFiles/CMakeError.log".
     917    ---------------------------------------------
  >> 918    Error when bootstrapping CMake:
     919    Problem while running initial CMake
     920    ---------------------------------------------
```

This happened when I specified the target as `x86_64`.  

It did not throw the error if I did not specify a target and just let spack choose (`haswell` in this case).

--- a/lib/spack/llnl/util/cpu/microarchitectures.json
+++ b/lib/spack/llnl/util/cpu/microarchitectures.json
@@ -49,11 +49,18 @@
       "vendor": "generic",
       "features": [],
       "compilers": {
-        "gcc": {
-          "versions": "4:",
-          "name": "x86-64",
-          "flags": "-march={name} -mtune={name}"
-        }
+        "gcc": [
+          {
+            "versions": "4.2.0:",
+            "name": "x86-64",
+            "flags": "-march={name} -mtune=generic"
+          },
+          {
+            "versions": "4.0:4.1.2",
+            "name": "x86-64",
+            "flags": "-march={name} -mtune={name}"
+          }
+        ]
       }
     },
     "nocona": {

--- a/lib/spack/spack/test/llnl/util/cpu.py
+++ b/lib/spack/spack/test/llnl/util/cpu.py
@@ -190,7 +190,9 @@ def test_target_json_schema():
 
 
 @pytest.mark.parametrize('target_name,compiler,version,expected_flags', [
-    ('x86_64', 'gcc', '4.9.3', '-march=x86-64 -mtune=x86-64'),
+    ('x86_64', 'gcc', '4.9.3', '-march=x86-64 -mtune=generic'),
+    ('x86_64', 'gcc', '4.2.0', '-march=x86-64 -mtune=generic'),
+    ('x86_64', 'gcc', '4.1.1', '-march=x86-64 -mtune=x86-64'),
     ('nocona', 'gcc', '4.9.3', '-march=nocona -mtune=nocona'),
     ('nehalem', 'gcc', '4.9.3', '-march=nehalem -mtune=nehalem'),
     ('nehalem', 'gcc', '4.8.5', '-march=corei7 -mtune=corei7'),

